projects/packages/jetpack-mu-wpcom/src/features/wpcom-media/wpcom-media-url-upload.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Allows uploading media from URL in Media Library.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * Appends the wpcom media URL upload form.
+ */
+function wpcom_media_url_upload() {
+	global $pagenow;
+
+	?>
+	<div id="wpcom-media-url-upload"></div>
+	<?php
+
+	$handle = jetpack_mu_wpcom_enqueue_assets( 'wpcom-media-url-upload', array( 'js', 'css' ) );
+
+	$data = wp_json_encode(
+		array(
+			'ajaxUrl'      => admin_url( 'admin-ajax.php' ),
+			'action'       => 'wpcom_media_url_upload',
+			'nonce'        => wp_create_nonce( 'wpcom_media_url_upload' ),
+			'isSiteEditor' => $pagenow === 'site-editor.php',
+		)
+	);
+
+	wp_add_inline_script(
+		$handle,
+		"window.JETPACK_MU_WPCOM_MEDIA_URL_UPLOAD = $data;",
+		'before'
+	);
+}
+
+/**
+ * AJAX handler for the wpcom media URL upload.
+ */
+function wpcom_handle_media_url_upload() {
+	check_ajax_referer( 'wpcom_media_url_upload' );
+
+	if ( ! isset( $_POST['url'] ) ) {
+		return;
+	}
+
+	$url = esc_url_raw( wp_unslash( $_POST['url'] ) );
+
+	$tmp = download_url( $url );
```

Comment:
nit: it would be better to call it something like `$tmp_file`

Reply:
Fixed via https://github.com/Automattic/jetpack/pull/41089/commits/8ed9c5b6daa59ce7473c802a2c02a0f0a202d7cb